modules/account-lib/src/coin/hbar/keyPair.ts

```diff
@@ -53,7 +55,10 @@ export class KeyPair extends Ed25519KeyPair {
   /** @inheritdoc */
   recordKeysFromPrivateKeyInProtocolFormat(prv: string): DefaultKeys {
     try {
-      const hederaPrv = Ed25519PrivateKey.fromString(prv);
+      if (prv.length % 2 !== 0) {

```

Comment:
Without this check, this step does not fail now. We have a unit test that was making sure that by adding an extra number at the end of the key, we failed.